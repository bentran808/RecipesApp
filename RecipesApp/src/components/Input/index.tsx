import React, { useCallback } from 'react';
import isEqual from 'react-fast-compare';
import { Image, ImageSourcePropType, Pressable, TextInput, View } from 'react-native';
import styles from './styles';

type Props = {
  keyword: string;
  onChangeText: (text: string) => void;
  leftIcon?: ImageSourcePropType;
  rightIcon?: ImageSourcePropType;
  onPressLeftIcon?: () => void;
  onPressRightIcon?: () => void;
};

const Input = ({
  keyword,
  onChangeText,
  leftIcon,
  rightIcon,
  onPressLeftIcon,
  onPressRightIcon
}: Props) => {
  const handlePressLeftIcon = useCallback(() => {
    onPressLeftIcon && onPressLeftIcon();
  }, []);

  const handlePressRightIcon = useCallback(() => {
    onPressRightIcon && onPressRightIcon();
  }, []);

  return (
    <View style={styles.searchContainer}>
      {leftIcon && (
        <Pressable testID='leftIcon' onPress={handlePressLeftIcon}>
          <Image style={styles.searchIcon} source={leftIcon} />
        </Pressable>
      )}
      <TextInput style={styles.searchInput} onChangeText={onChangeText} value={keyword} />
      {rightIcon && (
        <Pressable testID='rightIcon' onPress={handlePressRightIcon}>
          <Image style={styles.searchIcon} source={rightIcon} />
        </Pressable>
      )}
    </View>
  );
};

export default React.memo(Input, isEqual);
