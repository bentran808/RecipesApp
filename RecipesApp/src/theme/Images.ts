export const BackArrow = require('assets/images/icons/backArrow.png');
export const MenuIcon = require('assets/images/icons/menu.png');
export const HomeIcon = require('assets/images/icons/home.png');
export const CategoryIcon = require('assets/images/icons/category.png');
export const SearchIcon = require('assets/images/icons/search.png');
export const TimeIcon = require('assets/images/icons/time.png');
export const CloseIcon = require('assets/images/icons/close.png');
export const BasketIcon = require('assets/images/icons/basket.png');
export const CartIcon = require('assets/images/icons/cart.png');
