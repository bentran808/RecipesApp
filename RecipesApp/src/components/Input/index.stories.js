import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import CenterView from 'storybook/stories/CenterView';
import { CloseIcon, SearchIcon } from 'theme';
import Input from '.';

const handleChangeText = action('item-changed');
const handlePressLeftIcon = action('clicked-left-icon');
const handlePressRightIcon = action('clicked-right-icon');
storiesOf('Input', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('default', () => <Input keyword="" onChangeText={handleChangeText} />)
  .add('with left icon', () => (
    <Input
      keyword=""
      onChangeText={handleChangeText}
      leftIcon={SearchIcon}
      onPressLeftIcon={handlePressLeftIcon}
    />
  ))
  .add('with right icon', () => (
    <Input
      keyword=""
      onChangeText={handleChangeText}
      rightIcon={CloseIcon}
      onPressRightIcon={handlePressRightIcon}
    />
  ))
  .add('with left and right icon', () => (
    <Input
      keyword=""
      onChangeText={handleChangeText}
      leftIcon={SearchIcon}
      onPressLeftIcon={handlePressLeftIcon}
      rightIcon={CloseIcon}
      onPressRightIcon={handlePressRightIcon}
    />
  ));
