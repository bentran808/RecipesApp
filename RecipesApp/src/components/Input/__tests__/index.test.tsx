import Input from 'components/Input';
import { Pressable } from 'react-native';
import renderer from 'react-test-renderer';
import { CloseIcon, SearchIcon } from 'theme';

describe('Input Component', () => {
  const onChangeText = jest.fn();
  const onPressLeftIcon = jest.fn();
  const onPressRightIcon = jest.fn();
  const component = renderer.create(
    <Input
      keyword={''}
      onChangeText={onChangeText}
      leftIcon={SearchIcon}
      rightIcon={CloseIcon}
      onPressLeftIcon={onPressLeftIcon}
      onPressRightIcon={onPressRightIcon}
    />
  );
  test('should render correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call function handlePressLeftIcon', () => {
    const button = component.root.findAllByType(Pressable)[0];

    button.props.onPress();

    expect(onPressLeftIcon).toHaveBeenCalled();
  });

  test('should call function handlePressRightIcon', () => {
    const button = component.root.findAllByType(Pressable)[1];

    button.props.onPress();

    expect(onPressRightIcon).toHaveBeenCalled();
  });
});
