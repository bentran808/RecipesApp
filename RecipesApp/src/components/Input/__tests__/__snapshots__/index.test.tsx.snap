// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EDEDED",
      "borderRadius": 10,
      "flexDirection": "row",
      "justifyContent": "space-around",
      "width": 250,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="leftIcon"
  >
    <Image
      source={1}
      style={
        Object {
          "height": 20,
          "tintColor": "grey",
          "width": 20,
        }
      }
    />
  </View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[MockFunction]}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#EDEDED",
        "color": "black",
        "height": 50,
        "width": 180,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="rightIcon"
  >
    <Image
      source={1}
      style={
        Object {
          "height": 20,
          "tintColor": "grey",
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
